src/parsers/multitest.js

```diff
@@ -1,29 +1,26 @@
-const PROTRACTOR_SHARDED = /------------------------------------/g
-const SPECFILE_REG = /.+Specs:\s(.*\.js)/g
+const PROTRACTOR_MULTITEST = /------------------------------------/g
+const SPECFILE_REG = /Specs:\s(.*\.js)/g
 
 export default {
-  name: 'ShardedParser',
+  name: 'MultiTestParser',
   test (output) {
-    return PROTRACTOR_SHARDED.test(output) && SPECFILE_REG.test(output)
+    return PROTRACTOR_MULTITEST.test(output) && SPECFILE_REG.test(output)
   },
 
   parse (failedSpecs, output) {
     let match = null
-    let testsOutput = output.split('------------------------------------')
-    testsOutput.shift()
-    let RESULT_REG = /\d+\sspec|assertions?,\s(\d+)\sfailures?/g
+    let testsOutput = output.split('------------------------------------').slice(1)
+    let RESULT_REG = /,\s0 failures/g
     testsOutput.forEach(function (test) {
       let specfile
       let result = 'failed'
       // retrieve specfile from run;
       while (match = SPECFILE_REG.exec(test)) { // eslint-disable-line no-cond-assign
         specfile = match[1]
       }
-      // check for string 'X specs, X failures' and verify that failures === 0;
+      // check for string '0 failures' and then marks the test as passed
       while (match = RESULT_REG.exec(test)) { // eslint-disable-line no-cond-assign
```

Comment:
Ahhh, gotcha! Yeah the `No filename given` case is pretty difficult; I don't have any great ideas at the moment 😦 

It's pretty bad to silently swallow failures. Do you happen to have some example output with the `No filename given` case?

Also, could you just amend the last commit with the option b commit? I'd like to merge this in as a single commit if possible.